spec/models/activity_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+RSpec.describe Activity, type: :model do
+  it { is_expected.to belong_to(:actor).inverse_of(:activities).class_name(User.name) }
+
+  let(:instance) { create(:instance) }
+  with_tenant(:instance) do
+    describe '.create' do
+      let(:user) { create(:user, name: 'tester') }
+      let(:course) { create(:course) }
+      subject(:activity_and_notifications) do
+        Activity.new(actor: user, object: user, event: :tested).notify(user, type: :popup).
+          notify(course, type: :feed).save
+      end
+
+      it 'creates an activity' do
+        expect { activity_and_notifications }.to change { Activity.count }.by(1)
+        expect(Activity.find_by(actor: user).event).to eq('users.notifications.tested')
+      end
+
+      it 'creates a user notification' do
+        expect { activity_and_notifications }.to change { UserNotification.count }.by(1)
+      end
+
+      it 'creates a course notification' do
+        expect { activity_and_notifications }.to change { Course::Notification.count }.by(1)
+      end
+    end
+
+    describe '#notify' do
+      let(:user) { create(:user, name: 'tester') }
+      let(:course) { create(:course) }
+      let(:activity) do
+        create(:activity, actor: user, object: course, event: 'users.notifications.tested')
```

Comment:
isn't the event key generated when saved? won't the saved event key be duplicated?